core/schema/git.go

```diff
@@ -110,30 +123,41 @@ type tagArgs struct {
 
 func (s *gitSchema) tag(ctx context.Context, parent *core.GitRepository, args tagArgs) (*core.GitRef, error) {
 	return &core.GitRef{
-		Ref:  args.Name,
-		Repo: parent,
+		Query: parent.Query,
+		Ref:   args.Name,
+		Repo:  parent,
 	}, nil
 }
 
 type treeArgs struct {
-	// SSHKnownHosts is deprecated
-	SSHKnownHosts string `json:"sshKnownHosts"`
-	// SSHAuthSocket is deprecated
-	SSHAuthSocket socket.ID `json:"sshAuthSocket"`
+	SSHKnownHosts dagql.Optional[dagql.String]  `name:"sshKnownHosts" deprecated:"This option should be passed to git() instead."`
+	SSHAuthSocket dagql.Optional[core.SocketID] `name:"sshAuthSocket" deprecated:"This option should be passed to git() instead."`
```

Comment:
Somehow these `deprecated` options don't end up propagating into `dagger.gen.go`. These two fields also don't have docs in `Install`, so there's no information on them.

Reply:
good catch, taking a look!

Reply:
Looks like this wasn't making it to the client [before](https://github.com/dagger/dagger/blob/v0.9.6/sdk/go/dagger.gen.go#L2867-L2885), either; the [introspection query](https://github.com/dagger/dagger/blob/1c97437619ad23d656fade8986b087026e054cb1/cmd/codegen/introspection/introspection.graphql#L59-L66) it uses doesn't select the `isDeprecated` or `deprecationReason` fields.

Weirdly enough neither does the Apollo GraphQL sandbox explorer (left is v0.9.6, right is local dev):

![image](https://github.com/dagger/dagger/assets/1880/e80194af-c76d-4007-828b-10116f4d2a03)

(Placeholder docs added on the right just to suss out any "maybe it only shows directives if there are docs" nonsense)

Seems like we need to either update SDK codegen to look for `@deprecated` or manually set a description. The latter seems like less work for everyone in the grand scheme of things. Maybe setting something to deprecated could default its description to `DEPRECATED: <deprecation reason>`?